### PR TITLE
Update/use h5p settings

### PR DESCRIPTION
**Pull request by Bruswei**

### CODE REVIEW

File: H5P.d.ts [hunk 1]
```diff
@@ -1,62 +1,6 @@
...
-  findSemanticsField: (
-    fieldName: string,
-    semanticsStructure: H5PField | Array<H5PField>,
-  ) => H5PField | Array<H5PField> | null;
-}
+import { NDLATagsPicker } from './src/widgets/NDLATagsPicker.widget';
+import { H5PField } from './src/types/H5P/H5PField';
+import { H5PForm } from './src/types/H5P/H5PForm';
 
 declare class EventDispatcher {
```
Comment (boyum) [inline H5P.d.ts:5]: I think all of these can be imported from `h5p-types` instead.

File: package.json [hunk 2]
```diff
@@ -38,23 +35,31 @@
...
     "eslint-plugin-jsx-a11y": "^6.6.0",
     "eslint-plugin-react": "^7.30.1",
     "eslint-plugin-react-hooks": "^4.6.0",
+    "html-webpack-plugin": "^5.5.1",
     "identity-obj-proxy": "^3.0.0",
     "jest": "^29.4.3",
     "jest-environment-jsdom": "^29.4.3",
-    "prettier": "^2.7.1",
+    "mini-css-extract-plugin": "^2.7.6",
+    "node-sass": "^9.0.0",
```
Comment (boyum) [inline package.json:54]: Why do we need `node-sass`?

File: src/index.tsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
-import { H5PEditor } from "./H5P/H5P.util";
-import { NDLATagsPicker } from "./widgets/NDLATagsPicker.widget";
-import "./styles.scss";
+import { H5PEditor } from './H5P/H5P.util';
+import { NDLATagsPicker } from './widgets/NDLATagsPicker.widget';
+import './styles.scss';
 
 H5PEditor.widgets.NDLATagsPicker = NDLATagsPicker;
 H5PEditor.NDLATagsPicker = NDLATagsPicker;
```
Comment (boyum) [inline src/index.tsx:6]: `registerWidget('NDLATagsPicker', 'NDLATagsPicker', NDLATagsPicker)`, imported from `h5p-utils`

File: src/types/H5P/H5PBehaviour.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
-import { H5PFieldGroup } from "./H5PField";
+import { H5PFieldGroup } from './H5PField';
```
Comment (boyum) [inline src/types/H5P/H5PBehaviour.ts:1]:
```suggestion
import { H5PFieldGroup } from 'h5p-types';
```

File: package.json (hunk 2 above)
Comment (Bruswei) [reply package.json:54]: Not sure how I managed install node-sass, uninstalling.

**Comment by boyum** [comment]
Please update `.h5pignore` as well. We don't want to export any config files or TS files (`typings.d.ts` for instance).

File: package.json [hunk 5]
```diff
@@ -38,23 +35,30 @@
...
     "eslint": "^8.19.0",
     "eslint-config-airbnb": "^19.0.4",
-    "eslint-config-prettier": "^8.5.0",
+    "eslint-config-ndla-h5p": "github:NDLANO/eslint-config-ndla-h5p",
     "eslint-import-resolver-typescript": "^2.7.1",
     "eslint-plugin-import": "^2.26.0",
     "eslint-plugin-jsx-a11y": "^6.6.0",
     "eslint-plugin-react": "^7.30.1",
     "eslint-plugin-react-hooks": "^4.6.0",
+    "html-webpack-plugin": "^5.5.1",
```
[the remark this answers is not in the document]
Comment (Bruswei) [reply package.json:49]: A tool I tried out for optimization, left over and will be deleted.

File: src/types/H5P/H5PBehaviour.ts (hunk 4 above)
Comment (Bruswei) [reply src/types/H5P/H5PBehaviour.ts:1]: H5P-types and H5P-utils will be moved to a new PR

File: .h5pignore [hunk 6]
```diff
@@ -3,6 +3,8 @@ src
 .babelrc
 .gitignore
 .h5pignore
+.eslintignore
+.eslntrc.json
```
Comment (boyum) [inline .h5pignore:7]:
```suggestion
.eslintrc.json
```

File: .h5pignore [hunk 7]
```diff
@@ -12,3 +14,6 @@ storybook-static
 docs
 babel.config.js
 tsconfig.json
+LICENSE
```
Comment (boyum) [inline .h5pignore:17]: LICENSE should be part of the h5p package

```suggestion
```

File: .h5pignore [hunk 8]
```diff
@@ -12,3 +14,6 @@ storybook-static
...
+jest.config.ts
+typings.d.s
```
Comment (boyum) [inline .h5pignore:19]:
```suggestion
typings.d.ts
```

File: .h5pignore (hunk 6 above)
Comment (boyum) [inline .h5pignore:7]:
```suggestion
.eslintrc.json
```

File: .h5pignore (hunk 7 above)
Comment (otacke) [reply .h5pignore]: Sorry, LICENSE has to be excluded as well. H5P core will not allow files without any file extension.

If you want to keep the license information in the H5P library package, then you should rename it to LICENSE.txt which will also be identified by github automatically and would be allowed inside H5P library packages.

**Review by boyum — APPROVED** [review]

File: .h5pignore (hunk 8 above)
Comment (boyum) [reply .h5pignore]: ⚠️ Add `.github`, `build_info` and `h5p.d.ts` as well.

**Review by boyum — CHANGES_REQUESTED** [review]
Still a few missing h5pignore records

**Review by boyum — COMMENTED** [review]
The filenames for built files and the filenames in library.json don't match.

image
image

File: .h5pignore (hunk 8 above)
Comment (Bruswei) [reply .h5pignore]: Added.

**Comment by Bruswei** [comment]
> The filenames for built files and the filenames in library.json don't match.
> image image

Update, please double check :)

**Review by boyum — CHANGES_REQUESTED** [review]
h5pignore `h5p.d.ts` as well.

**Review by boyum — APPROVED** [review]

**Pull request merged**